executor/write_test.go

```diff
@@ -1330,7 +1330,26 @@ func (s *testSuite) TestUpdate(c *C) {
 	tk.MustExec("update t set a = ''")
 	tk.MustQuery("select * from t").Check(testkit.Rows("0000-00-00 00:00:00 1999-12-13 00:00:00"))
 	tk.MustExec("update t set b = ''")
-	tk.MustQuery("select * from t").Check(testkit.Rows("0000-00-00 00:00:00 <nil>"))
+	tk.MustQuery("select * from t").Check(testkit.Rows("0000-00-00 00:00:00 0000-00-00 00:00:00"))
```

Comment:
I was wondering if I could know why we need change this line?

Reply:
It is consistent with MySQL now. The following is the result of MySQL:
```sql
mysql> create table t(a datetime not null, b datetime);
Query OK, 0 rows affected (0.33 sec)

mysql> insert into t value('1999-12-12', '1999-12-13') ;
Query OK, 1 row affected (0.13 sec)

mysql> set sql_mode='';
Query OK, 0 rows affected, 1 warning (0.00 sec)

mysql> update t set b = '';
Query OK, 1 row affected, 1 warning (0.12 sec)
Rows matched: 1  Changed: 1  Warnings: 1

mysql> select * from t;
+---------------------+---------------------+
| a                   | b                   |
+---------------------+---------------------+
| 1999-12-12 00:00:00 | 0000-00-00 00:00:00 |
+---------------------+---------------------+
1 row in set (0.00 sec)
```